tests/e2e/nodes.go

```diff
@@ -44,4 +47,60 @@ var _ = Describe("[cloud-provider-aws-e2e] nodes", func() {
 			gomega.Expect(node.Labels).To(gomega.HaveKey("topology.k8s.aws/zone-id"))
 		}
 	})
+
+	It("should label nodes with topology network info if instance is supported", func(ctx context.Context) {
+		framework.ExpectNoError(e2enode.WaitForAllNodesSchedulable(f.ClientSet, 10*time.Minute))
+		nodeList, err := e2enode.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err)
+
+		if len(nodeList.Items) < 2 {
+			framework.Failf("Conformance requires at least two nodes")
+		}
+		clientConfig, err := framework.LoadConfig()
+		framework.ExpectNoError(err)
+		client, err := kubernetes.NewForConfig(clientConfig)
+		framework.ExpectNoError(err)
+
+		ssar := &authv1.SelfSubjectAccessReview{
+			Spec: authv1.SelfSubjectAccessReviewSpec{
+				ResourceAttributes: &authv1.ResourceAttributes{
+					Group:    "ec2.amazonaws.com",
+					Resource: "describeInstanceTopology",
+					Verb:     "get",
+				},
+			},
+		}
+		result, err := client.AuthorizationV1().SelfSubjectAccessReviews().Create(context.TODO(), ssar, metav1.CreateOptions{})
+		if err != nil {
+			framework.Failf("Error checking EC2 describeInstanceTopology access: %v", err)
+		}
+		allowed := result.Status.Allowed
+		supportedInstanceType := "p4d.24xlarge"
```

Comment:
nit: May be brittle if topology information is ever backported to vanilla instances, though would be easy enough to fix.